src/app/hooks/useVMWareSettings.tsx

```diff
@@ -0,0 +1,54 @@
+import { useMutation, useQueryClient } from '@tanstack/react-query';
+import { useDemoMode } from './useDemoMode';
+import { Meta } from '../types/Meta';
+import { updateMockData } from '../components/demo/update-mock-data';
+import { settingsObjectsClient, SettingsObjectCreate } from '@dynatrace-sdk/client-classic-environment-v2';
+import { VmwareSetting } from '../types/CloudTypes';
+import { VMWARE_SETTINGS_SCHEMA } from "../constants";
+
+async function noop() {
+  return Promise.resolve() as unknown as Promise<Response>;
+}
+
+async function fetcher(formData: FormData): Promise<unknown> {
+  const vmwareSettings : VmwareSetting = {
+    enabled: true,
+    label: formData.get('label')?.toString() || "",
+    ipaddress: formData.get('ipaddress')?.toString() || "",
+    username: formData.get('username')?.toString() || "",
+    password: formData.get('password')?.toString() || "",
+  }
+
+  const vmwarePayload = {
+    schemaId: VMWARE_SETTINGS_SCHEMA,
+    value: vmwareSettings,
+    scope: "environment",
+  } as SettingsObjectCreate;
+
+  return settingsObjectsClient.postSettingsObjects({body: [vmwarePayload]});
```

Comment:
```suggestion
  await settingsObjectsClient.postSettingsObjects({body: [vmwarePayload]});
```
You don't need the data returned. The error of the compiler comes from the noop function returning something different than the actual implementation.